landlab/components/flow_accum/lossy_flow_accumulator.py

```diff
@@ -0,0 +1,464 @@
+#!/usr/env/python
+
+"""
+lossy_flow_accumulator.py: Component to accumulate flow and calc drainage area,
+while permitting gain or loss of discharge during flow.
+
+DEJH, late 2018
+"""
+
+import sys
+
+from landlab.components.flow_accum import (
+    FlowAccumulator,
+    flow_accum_bw,
+    flow_accum_to_n,
+)
+
+if sys.version_info[0] >= 3:
+    from inspect import signature
+
+
+class LossyFlowAccumulator(FlowAccumulator):
+
+    """
+    Component to calculate drainage area and accumulate flow, while permitting
+    dynamic loss or gain of flow downstream.
+
+    This component is closely related to the FlowAccumulator, in that
+    this is accomplished by first finding flow directions by a user-specified
+    method and then calculating the drainage area and discharge. However,
+    this component additionally requires the passing of a function that
+    describes how discharge is lost or gained downstream,
+    f(Qw, nodeID, linkID, grid). See the Examples below to see how this works
+    in practice.
+
+    Optionally, spatially variable runoff can be set either by the model grid
+    field 'water__unit_flux_in' or the input variable *runoff_rate**.
+
+    Optionally a depression finding component can be specified and flow
+    directing, depression finding, and flow routing can all be accomplished
+    together. Note that the DepressionFinderAndRouter is not particularly
+    intelligent when running on lossy streams, and in particular, it will
+    reroute flow around pits even when they are in fact not filled due to loss.
+
+    NOTE: The perimeter nodes  NEVER contribute to the accumulating flux, even
+    if the  gradients from them point inwards to the main body of the grid.
+    This is because under Landlab definitions, perimeter nodes lack cells, so
+    cannot accumulate any discharge.
+
+    LossyFlowAccumulator stores as ModelGrid fields:
+
+        -  Node array of drainage areas: *'drainage_area'*
+        -  Node array of discharges: *'surface_water__discharge'*
+        -  Node array of discharge loss in transit (vol/sec). This is the
+            total loss across all of the downstream links:
+            *'surface_water__discharge_loss'*
+        -  Node array containing downstream-to-upstream ordered list of node
```

Comment:
@SiccarPoint lets have a group discussion about this field name. 

I like it, but I think its worth talking about it before we add a new standard name.

I've added it to the call agenda.

Reply:
I'll leave this until Monday...

Reply:
We're OK with this naming on the call.